app/script_variant.py

```diff
@@ -65,6 +69,46 @@ def romanize_japanese_name_by_name_dict(word):
     return [word]
 
 
+def romanize_japanese_name_by_name_dict(word, for_index=True):
+    """
+    This method romanizes japanese name by using name dictionary.
+    If word isn't found in dictionary, this method doesn't
+    apply romanize.
+    This method can return multiple romanizations.
+    (because there are multiple ways to read the same kanji name in japanese)
+    Args:
+        for_index: this method is called for indexing or not
+    Returns:
+        [romanized_jp_name, and romanized_jp_name(split_word), ...]
+    """
+    if not word:
+        return ['']
+
+    words = []
+    for index in xrange(1, len(word)):
+        # split word because query word may not contains white space
+        # e.g., if the query (e.g., "山田") doesn't contain white space,
+        # it would work return words ("yamata, yamada")
+        first_part = word[:index]
+        last_part = word[index:]
+        romanized_first_parts = romanize_single_japanese_word_by_name_dict(
+            first_part)
+        romanized_last_parts = romanize_single_japanese_word_by_name_dict(
+            last_part)
+        for romanized_first_part in romanized_first_parts:
+            for romanized_last_part in romanized_last_parts:
+                if romanized_first_part != first_part and \
+                        romanized_last_part != last_part:
+                    words.append(romanized_first_part+romanized_last_part)
+                    # To search by the query which doesn't contains white sapce,
```

Comment:
This comment looks wrong. If the query is a full name without white space e.g., "やまだたろう", we don't need these fields, right? You need to explain:

- Why you need to add these fields in index
- Why you must not add these fields for search queries

It may be easier to explain using an example.